gadgets/trace_tcpconnect/test/trace_tcpconnect_test.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/containers"
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+	"github.com/stretchr/testify/require"
+)
+
+type traceTcpconnectEvent struct {
+	eventtypes.CommonData
+
+	Timestamp string `json:"timestamp"`
+	MntNsID   uint64 `json:"mntns_id"`
+
+	Comm string `json:"comm"`
+	Pid  uint32 `json:"pid"`
+	Tid  uint32 `json:"tid"`
+	Uid  uint32 `json:"uid"`
+	Gid  uint32 `json:"gid"`
+
+	Latency     uint64           `json:"latency,omitempty"`
+	SrcEndpoint utils.L4Endpoint `json:"src"`
+	DstEndpoint utils.L4Endpoint `json:"dst"`
+}
+
+func TestTraceTcpconnect(t *testing.T) {
+	gadgettesting.RequireEnvironmentVariables(t)
+	utils.InitTest(t)
+
+	containerFactory, err := containers.NewContainerFactory(utils.Runtime)
+	require.NoError(t, err, "new container factory")
+	containerName := "test-trace-tcpconnect"
+	containerImage := "docker.io/library/nginx:latest"
+
+	var ns string
+	containerOpts := []containers.ContainerOption{containers.WithContainerImage(containerImage)}
+	if utils.CurrentTestComponent == utils.KubectlGadgetTestComponent {
+		ns = utils.GenerateTestNamespaceName(t, "test-trace-tcpconnect")
+		containerOpts = append(containerOpts, containers.WithContainerNamespace(ns))
+	}
+	cmds := "nginx && while true; do curl 127.0.0.1; sleep 0.1; done"
+	testContainer := containerFactory.NewContainer(containerName, cmds, containerOpts...)
+	testContainer.Start(t)
+	t.Cleanup(func() {
+		testContainer.Stop(t)
+	})
+
+	var runnerOpts []igrunner.Option
+	var testingOpts []igtesting.Option
+	commonDataOpts := []utils.CommonDataOption{utils.WithContainerImageName(containerImage), utils.WithContainerID(testContainer.ID())}
+
+	switch utils.CurrentTestComponent {
+	case utils.IgLocalTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-r=%s", utils.Runtime)))
+	case utils.KubectlGadgetTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-n=%s", ns)))
+		testingOpts = append(testingOpts, igtesting.WithCbBeforeCleanup(utils.PrintLogsFn(ns)))
+		commonDataOpts = append(commonDataOpts, utils.WithK8sNamespace(ns))
+	}
+
+	runnerOpts = append(runnerOpts, igrunner.WithValidateOutput(
+		func(t *testing.T, output string) {
+			expectedEntries := &traceTcpconnectEvent{
+				CommonData: utils.BuildCommonData(containerName, commonDataOpts...),
+				SrcEndpoint: utils.L4Endpoint{
+					Addr:    "127.0.0.1",
+					Version: 4,
+					Port:    utils.NormalizedInt,
+					Proto:   6,
+				},
+				DstEndpoint: utils.L4Endpoint{
+					Addr:    "127.0.0.1",
+					Version: 4,
+					Port:    utils.NormalizedInt,
+					Proto:   6,
+				},
+				Comm: "curl",
+				Uid:  0,
+				Gid:  0,
```

Comment:
Since we are not using `setuidgid` here in the command I would just set them to `utils.NormalizedInt`